### DISCUSSION TITLE
Song view : enhance the live performance

### DISCUSSION THREAD

**Discussion by AtraWanderer** [SynthstromAudible/DelugeFirmware#700] [Ideas]
A great implementation would be to have a fluid workflow on live perf : 

While playing a song with some clips on : on a song change the remaining clips may be deported to the next load.
In the new song whenever you mute the "old" clips, they are gone forerver (memory issues?).
Maybe the "old" clips should be with a specific color (with the clip groups) at the top of the song mode view.

![deluge](https://github.com/SynthstromAudible/DelugeFirmware/assets/11061751/b5a9cd9c-54ff-47e1-9513-38ef8ba70e5f)

The goal of this is to make the mix in live perf really more easy and with a lot more adptability within the song chosen.

**Comment by AtraWanderer** [comment]
I would be really interested to know what you guys think about this ? One of the feature taht would really improve the deluge workflow.
Tell me if it's not wished, etc?

**Reply by sapphire-arches** [reply]
this is an interesting idea. A few considerations:
  - If the old clips are never removed or are edited, do you expect them to save with the old song or the new? The easiest (and maybe only possible) implementation is to save with the new song
  - Should old clips be deleted on play head stop, or only on mute?
  - How should the pad grid indicate that these clips are going to be deleted on mute?

The heart of the idea, keeping clips across songs, is definitely interesting.  The implementation and UI could be challenging however.

**Reply by AtraWanderer** [reply]
- I was more seeing it as something temporary, just for live convinient, and so never beeing saved in the new one, as if they didnt exist in it. It's more of a "DJ Mixing" style of deluge.
- Beeing deleted as long as they aren't clips from the original song. Myabe someone would have an interest of keeping it, tell me?
- Maybe a specific color on the "groups" section to specify they are old ones. Or, I don't kow if it's implemented : are the mute (green red yellow) able to display another color? Maybe when play they should have a specific green so you know it'ss temporary?
